tool/clean/clean_file_system/clean_file_system.go

```diff
@@ -0,0 +1,82 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+package main
+
+import (
+	"context"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/efs"
+	"log"
+	"time"
+)
+
+const daysToKeep = 30
+const keepDuration = -1 * time.Hour * 24 * time.Duration(daysToKeep)
+
+func main()  {
+	log.Printf("Begin to clean EFS resources")
+	expirationDate := time.Now().UTC().Add(keepDuration)
+	cxt := context.Background()
+	defaultConfig, err := config.LoadDefaultConfig(cxt)
+	if err != nil {
+		log.Fatalf("Error getting default config %v", err)
+	}
+	efsclient := efs.NewFromConfig(defaultConfig)
+
+	//get efs to delete
+	var nextToken *string
+	for {
+		describeFileSystemsInput := efs.DescribeFileSystemsInput{Marker: nextToken}
+		describeFileSystemsOutput, err := efsclient.DescribeFileSystems(cxt, &describeFileSystemsInput)
+		if err != nil {
+			log.Fatalf("Err %v", err)
+		}
+		for _, fileSystem := range describeFileSystemsOutput.FileSystems {
+			if expirationDate.After(*fileSystem.CreationTime) {
+				log.Printf("Trying to delete file system %s launch-date %v", *fileSystem.FileSystemId, fileSystem.CreationTime)
+				if fileSystem.NumberOfMountTargets > 0 {
+					err = deleteMountTargets(cxt, efsclient, fileSystem.FileSystemId)
```

Comment:
Personally I'd prefer just paginating and gathering all of the IDs at once, and then deleting at the end. There's a lot of nesting here. Just a nitpick.

Reply:
I'm fine with that. ECS has a batch of deleting but boo for not the same as EFS.